### DISCUSSION TITLE
OpenShift Extensibility

### DISCUSSION THREAD

**Discussion by neilchaudhuri** [cue-lang/cue#1788] [General]
There has been a lot of config drift on my OpenShift project, and I would like to understand how I can use Cue to typecheck our YAML against OpenShift objects (superset of Kubernetes objects) for the right data types but also for missing values or nonsense values like "TBD."

Do I need to create my own OpenShift type definitions? Are there other tips or tutorials anyone can suggest? Any help is appreciated.

Thanks.

**Comment by rogpeppe** [comment]
This seems more like a discussion than an issue to me, as there's no actual potential bug or feature request here AFAICS, so converting it to a discussion so we can talk about it there.

**Comment by PierreR** [comment]
Hello @neilchaudhuri we are using CUE to configure all the admin resources of our OpenShift clusters.

You don't need to create your own definition. You can simply import them.

```
go mod download github.com/openshift/api
cue get go github.com/openshift/api/route/v1
```

**Comment by neilchaudhuri** [comment]
This is helpful. Thanks.
